### PR TITLE
Implement some tests for request handler

### PR DESCRIPTION
**Pull request by andyg7**
Hey guys,

1) I wrote a few tests here for request_handler.py, the new module I refactored out. As of now I just added tests for three methods from the RequestHandler class to get some feedback. I used MagicMock to mock out the api calls, and basically ran a few tests to make sure the proper code was called. Was looking just to get some code coverage going.

 2) I also simplified some of the tests from test_load_data.py as per the discussion from #110 that @Saturn suggested.

3) I also added mock==1.0.1 to the requirements.txt file.

4) As mentioned in a previous PR to run all tests run the following command: "python -m unittest discover tests" from the root of the repo. This will go into tests/ and look to run __all__ tests

Thanks for any feedback :)
Andy

### CODE REVIEW

File: tests/test_request_handler.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    def test_bad_code(self, mock_call):
+        mock_call.return_value = mock.MagicMock(
+                status_code=requests.codes.bad,
+                response=json.dumps({'key': 'value'}))
+        raised = False
+        try:
+            self.rq._get(self.dummy_url)
+        except APIErrorException:
+            raised = True
+        self.assertTrue(raised is True)
```
Comment (carlosvargas) [inline tests/test_request_handler.py]: For this and other response code tests, we should assert the message we output.

**Comment by andyg7** [comment]
Hey @carlosvargas,
Thanks a lot for the feedback. I replaced the raised checks with a assertFails in all cases, and also added some checks on the error messages that are reported when an exception is.
Let me know of any more feedback.
Thanks so much,
Andy

File: tests/test_request_handler.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    @mock.patch('requests.get')
+    def test_get_team_scores_ok(self,
+                                mock_request_call, mock_writer):
+        mock_request_call.side_effect = [mocked_requests_get({'fixtures': [1, 2]}, 200)]
+        mock_writer.return_value = mock.Mock()
+        self.rq.get_team_scores("AFC", 6, True, True)
+        mock_writer.assert_called_once()
+
+    @mock.patch('test_request_handler.click.secho')
+    @mock.patch('test_request_handler.Stdout.team_scores')
```
Comment (Saturn) [inline tests/test_request_handler.py]: Is there a reason this cannot be `@mock.patch('soccer.writer.Stdout.team_scores')` and `@mock.patch('click.secho')` etc?

Comment (andyg7) [inline tests/test_request_handler.py]: Ah good call thanks - looks like I can do what you mentioned. Will fix now.

**Comment by andyg7** [comment]
Hey guys, should I squash my commits whenever I make a change? I worry that re-writing history could mess things up (even though it's my forked version) but it also seems annoying to have a lot of commits with tiny changes.

**Comment by carlosvargas** [comment]
I personally like to do that on PRs, but it's not a big deal.

Everything looks good to me 👍 Let's wait on @Saturn's input.

**Comment by Saturn** [comment]
Looks good.

**Pull request merged**